crates/compiler/can/src/expr.rs

```diff
@@ -2377,10 +2387,11 @@ pub fn inline_calls(var_store: &mut VarStore, expr: Expr) -> Expr {
         }
 
         Call(boxed_tuple, args, called_via) => {
-            let (fn_var, loc_expr, closure_var, expr_var) = *boxed_tuple;
+            let (fn_var, loc_expr, closure_var, expr_var, fx_var) = *boxed_tuple;
 
             match loc_expr.value {
                 Var(symbol, _) if symbol.is_builtin() => {
+                    // NOTE: This assumes builtins are not effectful!
```

Comment:
Didn't you add `List.walk!`? Would that break this?

Maybe what you mean is that builtins don't use `Stmt(_)`, which is why you removed the `!` usage from one of the `Task` functions?

Reply:
This comment refers to builtins that would produce the effect, not to higher-order functions that would just call it. The latter should be fine because when inlined, they would still include the call to the real effectful function and the enclosing function would be properly categorized as such.